Implemented page profile

Max2002 opened the pull request.

Max2002 committed:
Implemented page profile

Max2002 committed:
Added color and background for nationality block in profile

Max2002 committed:
Changed color in address line

Max2002 committed:
Created component for copy element. Fixed warnings with props.
Updated axios.

GCoder-13 commented on src/App.js:
```diff
@@ -4,7 +4,7 @@ import { useEffect } from 'react';
 import Header from './sections/Header';
 import Home from './sections/Home';
-import Profile from './sections/Profile/indxe';
+import Profile from './sections/Profile/index';
```
./sections/Profile

GCoder-13 commented on src/assets/styles/variables.scss:
```diff
@@ -3,8 +3,10 @@ $fontSizeSmall: 10px;
+$fontSizeXL: 24px;
 $errorMessage: 26px;
+$fontSizeTitle: 32px;
+$error404: 80px;
```
Use variable font sizes only where they can be repeated

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
```
dont use libs

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,57 @@
+  return (
+    <div className={st.contact}>
+      {renderPrefix()}
+      {isLink ? (
```
???

Max2002 replied:
incorrect name variable

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,57 @@
+import { CopiedSvg, CopySvg } from '../../assets/icons';
+import st from './styles.module.scss';
+
+export default function CopyElement({ children, isLink, prefixLink }) {
```
Don’t use children for string

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,57 @@
+
+  return (
+    <div className={st.contact}>
+      {renderPrefix()}
```
???

GCoder-13 commented on src/components/CopyElement/styles.module.scss:
```diff
@@ -0,0 +1,62 @@
+  margin: 10px 0;
+  @include flex-vertical-centre;
+
+  a {
```
???

Max2002 replied:
don't access the selector directly

GCoder-13 commented on src/sections/Profile/index.js:
```diff
@@ -0,0 +1,77 @@
+  const { picture, fullName, age, email, phone, address, nat } =
+    useSelector(myProfileSelector);
+
+  const renderElement = (element, flag, width = 375, height = 15) => {
```
create component

GCoder-13 commented on src/sections/Profile/styles.module.scss:
```diff
@@ -0,0 +1,73 @@
+  }
+}
+
+.personalInfo {
```
????

Max2002 replied:
empty css class

Max2002 committed:
Changed copyToClipboard react to js copy and several small changes

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,53 @@
+  };
+
+  useEffect(() => {
+    const interval = setInterval(() => setIsCopied(false), 2000);
```
Infinity interval

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,53 @@
+        {isCopied ? (
+          <CopiedSvg className={st.copied} />
+        ) : (
+          <CopySvg onClick={() => handleCopy(content)} />
```
???

Max2002 replied:
it is neccessary, because value in variable content

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,53 @@
+        )}
+      </div>
+      {link ? (
+        <a className={st.link} href={`${prefixLink}${content}`}>
```
href

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,53 @@
+
+CopyElement.propTypes = {
+  content: PropTypes.string,
+  link: PropTypes.bool,
```
incorect type

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,53 @@
+CopyElement.propTypes = {
+  content: PropTypes.string,
+  link: PropTypes.bool,
+  prefixLink: PropTypes.string,
```
delete

GCoder-13 commented on src/components/CopyElement/styles.module.scss:
```diff
@@ -0,0 +1,62 @@
+  margin: 10px 0;
+  @include flex-vertical-centre;
+
+  svg {
```
???

Max2002 replied:
use css class

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,53 @@
+
+  return (
+    <div className={st.contact}>
+      <div className={st.icon}>
```
rename

GCoder-13 commented on src/components/LoadingElement/index.js:
```diff
@@ -0,0 +1,28 @@
+import st from '../../sections/Profile/styles.module.scss';
+
+export default function LoadingElement({ element, flag, width, height }) {
+  return flag ? (
```
rename flag

GCoder-13 commented on src/components/LoadingElement/index.js:
```diff
@@ -0,0 +1,28 @@
+
+export default function LoadingElement({ element, flag, width, height }) {
+  return flag ? (
+    element
```
???

GCoder-13 commented on src/sections/Profile/index.js:
```diff
@@ -0,0 +1,72 @@
+    useSelector(myProfileSelector);
+
+  return (
+    <div className={`container ${st.profile}`}>
```
container must be component

GCoder-13 commented on src/sections/Profile/styles.module.scss:
```diff
@@ -0,0 +1,70 @@
+    text-align: center;
+  }
+
+  span {
```
???

Max2002 replied:
use css class

Max2002 committed:
Changed infinity interval in useEffect file CopyElement.
Added component for container.
Changed prop element to children in file LoadingElement.

GCoder-13 commented on src/components/CopyElement/styles.module.scss:
```diff
@@ -0,0 +1,62 @@
+  @include flex-vertical-centre;
+}
+
+.copy, .copied {
```
use 1 class

GCoder-13 commented on src/sections/Profile/index.js:
```diff
@@ -0,0 +1,62 @@
+
+  return (
+    <Container>
+      <div className={st.profile}>
```
unnecessary wrapper

Max2002 committed:
Added prop className for component Container and several changes in copyElement style

GCoder-13 commented on src/components/Container/index.js:
```diff
@@ -0,0 +1,12 @@
+
+Container.propTypes = {
+  children: PropTypes.node.isRequired,
+  className: PropTypes.string.isRequired,
```
Not required

GCoder-13 commented on src/components/CopyElement/index.js:
```diff
@@ -0,0 +1,57 @@
+          {isCopied ? 'Copied' : 'Copy'}
+        </div>
+        {isCopied ? (
+          <CopiedSvg className={`${st.copiesSvg} ${st.copied}`} />
```
Clsx

GCoder-13 commented on src/components/CopyElement/styles.module.scss:
```diff
@@ -0,0 +1,62 @@
+    position: absolute;
+    bottom: -45%;
+    left: 37%;
+    border: 7px solid transparent;
```
Why?

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -24,32 +24,32 @@ export default function Menu(props) {
-      {loading ? (
-        <Skeleton width={200} height={20} />
-      ) : (
+      <LoadingElement loading={!loading} width={200}>
```
Loading?

Max2002 committed:
Changed flag loading in component LoadingElement and added component Hint

GCoder-13 commented on src/components/CopyElement/styles.module.scss:
```diff
@@ -0,0 +1,46 @@
+
+.copiesSvg {
+  font-size: $fontSizeLarge;
+  //margin-right: 10px;
```
delete

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -24,32 +24,32 @@ export default function Menu(props) {
         <img className={st.avatar} src={avatar} alt={label} />
-      )}
+      </LoadingElement>
       <div className={st.dropDownList}>
```
do you really need this container?

Max2002 replied:
no

GCoder-13 commented on src/components/Menu/index.js:
```diff
@@ -24,32 +24,32 @@ export default function Menu(props) {
 Menu.propTypes = {
   label: PropTypes.string.isRequired,
-  avatar: PropTypes.string.isRequired,
+  avatar: PropTypes.string,
```
avatarUrl

GCoder-13 commented on src/sections/Profile/index.js:
```diff
@@ -0,0 +1,60 @@
+        <LoadingElement loading={!picture.large} width={260} height={260}>
+          <img className={st.avatar} src={picture.large} alt={fullName} />
+        </LoadingElement>
+        <div>
```
do you really need this container?

Max2002 replied:
yes, because without this doesn`t work flex

GCoder-13 commented on src/sections/Profile/styles.module.scss:
```diff
@@ -0,0 +1,70 @@
+}
+
+.age {
+  font-size: 80%;
```
don't use percent for font-size. Only px or rem

Max2002 committed:
Several small changes in styles and one rename prop in Menu.js

GCoder-13 commented on src/components/Hint/index.js:
```diff
@@ -0,0 +1,16 @@
+import clsx from 'clsx';
+import st from './styles.module.scss';
+
+export default function Hint({ children, className }) {
```
hover effect must be implemented inside this component

Max2002 committed:
Removed hover effect hint from copyElement to component hint.
